src/main/java/org/opentripplanner/netex/mapping/StopMapper.java

```diff
@@ -44,23 +52,98 @@ Stop mapQuayToStop(
-        null,
-        fareZones,
-        null,
-        null,
-        transitMode.first,
-        transitMode.second
+            idFactory.createId(quay.getId()),
+            parentStation.getName(),
+            quay.getPublicCode(),
+            quay.getDescription() != null ? quay.getDescription().getValue() : null,
+            WgsCoordinateMapper.mapToDomain(quay.getCentroid()),
+            wheelchairBoarding,
+            null,
+            null,
+            fareZones,
+            null,
+            null,
+            transitMode.first,
+            transitMode.second
     );
+
     stop.setParentStation(parentStation);
 
     return stop;
   }
+
+  /**
+   * Get WheelChairBoarding from Quay and parent Station.
+   *
+   * @param quay          NeTEx quay could contain information about accessability
+   * @param parentStation Used to get a default WheelChairBoarding if not found from Quay.
+   * @return not null value with default NO_INFORMATION if nothing defined in quay or
+   * parentStation.
+   */
+  private WheelChairBoarding wheelChairBoardingFromQuay(Quay quay, Station parentStation) {
+
+    var defaultWheelChairBoarding = WheelChairBoarding.NO_INFORMATION;
+    var stopPlaceId = parentStation.getId();
+    var stopPlace = stopPlaceIndex.lookupLastVersionById(stopPlaceId.getId());
+
+    if (stopPlace != null) {
+      defaultWheelChairBoarding = wheelChairBoarding(
+              stopPlace.getAccessibilityAssessment(),
+              WheelChairBoarding.NO_INFORMATION
+      );
+    }
+
+    return wheelChairBoarding(quay.getAccessibilityAssessment(), defaultWheelChairBoarding);
+  }
+
+  /**
+   * If input and containing objects are not null, get the LimitationStatusEnumeration and map to
+   * internal {@link WheelChairBoarding} enumeration.
+   *
+   * @param accessibilityAssessment NeTEx object wrapping information regarding
+   *                                WheelChairBoarding
+   * @param defaultValue            If no {@link AccessibilityAssessment} is defined, default to
+   *                                this value
+   * @return Mapped enumerator, {@link WheelChairBoarding#NO_INFORMATION} if no value is found
+   */
+  private WheelChairBoarding wheelChairBoarding(
+          AccessibilityAssessment accessibilityAssessment,
+          WheelChairBoarding defaultValue
+  ) {
+    if (defaultValue == null) {
+      defaultValue = WheelChairBoarding.NO_INFORMATION;
+    }
+
+    if (accessibilityAssessment == null || accessibilityAssessment.getLimitations() == null
+            || accessibilityAssessment.getLimitations().getAccessibilityLimitation() == null ||
+            accessibilityAssessment.getLimitations()
+                    .getAccessibilityLimitation()
+                    .getWheelchairAccess() == null || accessibilityAssessment.getLimitations()
+            .getAccessibilityLimitation()
+            .getWheelchairAccess()
+            .value() == null) {
+
+      return defaultValue;
+
+    }
```

Comment:
I think it would be much more readable to use a Optional chain here

Reply:
Done